kivy/graphics/vertex_instructions_line.pxi

```diff
@@ -939,21 +970,34 @@ cdef class Line(VertexInstruction):
 
     cdef void prebuild_ellipse(self):
         cdef double x, y, w, h, angle_start = 0, angle_end = 360
-        cdef int angle_dir, segments = 0
+        cdef int angle_dir, segments = 0, extra_segments = 0
         cdef double angle_range
         cdef tuple args = self._mode_args
+        cdef int close_from_center = self._close_from_center
+
+        extra_segments = 3 if close_from_center else 1
 
         if len(args) == 4:
             x, y, w, h = args
         elif len(args) == 6:
             x, y, w, h, angle_start, angle_end = args
         elif len(args) == 7:
             x, y, w, h, angle_start, angle_end, segments = args
-            segments += 2
         else:
             x = y = w = h = 0
             assert 0
 
+        if 0 in (w, h):
+            return
+
+        if segments == 0 or segments < 3:
+            if segments != 0:
+                Logger.warning(f'{self.__class__.__name__} - ellipse: Number of segments not supported, they will be automatically set.')
+            segments = int(abs(angle_end - angle_start) / 2) + extra_segments
```

Comment:
Can we move this change (and related ones) into a separate PR?

Reply:
Hi @misl6!

Are you referring to reverting the changes from my last commit? And what other changes?